### PR TITLE
dialects: Add memref.dma_start op

### PR DESCRIPTION
**Pull request by AntonLydike**

### CODE REVIEW

**Review by webmiche — COMMENTED** [review]
Interoperation tests) :upside_down_face:

**Comment by AntonLydike** [comment]
Here are the docs for it: https://mlir.llvm.org/docs/Dialects/MemRef/#memrefdma_start-mlirmemrefdmastartop

**Comment by webmiche** [comment]
Okay this does indeed look weird. Not sure we should do something that they don't do in a mirrored dialect though...
Or does this op actually exist and look like this but the doc is just not there?

**Comment by AntonLydike** [comment]
image


This is the tablegen, I think we are right? This looks really crappy to be honest.

**Review by superlopuh — APPROVED** [review]

**Comment by webmiche** [comment]
In that case we can have my beloved interoperation tests, right?

**Comment by AntonLydike** [comment]
@webmiche it actually worked, I'm amazed!

File: tests/dialects/test_memref.py [hunk 1]
```diff
@@ -293,3 +297,61 @@ def test_memref_cast():
...
+    num_elements = TestSSAValue(IndexType())
+
+    dma_start = DmaStartOp.get(
+        src, [index, index], dest, [index], num_elements, tag, [index]
+    )
+
+    dma_start.verify()
+
+    # check that src index count is verified
+    with pytest.raises(VerifyException):
```
Comment (webmiche) [inline tests/dialects/test_memref.py:322]: (Just to mention it) You could check the error message here.

Comment (AntonLydike) [reply tests/dialects/test_memref.py]: Done! You can actually pass `match="..."` to the `pytest.raises` thingy to match, which makes it really nice!

**Review by webmiche — APPROVED** [review]

**Pull request merged**